theano/tensor/nnet/abstract_conv.py

```diff
@@ -66,7 +66,8 @@ def get_conv_output_shape(image_shape, kernel_shape,
     """
     bsize, imshp = image_shape[0], image_shape[2:]
     nkern, kshp = kernel_shape[0], kernel_shape[2:]
-
```

Comment:
I would be more strict here in the condition to make the graph simpler when nkern is symbolic. I think this condition would do it: "if nkern is not None and num_groups > 1".

This will probably fix the error in travis.

Some explanation of the travis error. Some part of Theano are not garanted to finish. So we added some check so if it iterate too much, it stop iterating instead of potentially never finishing. The threshold where we do this detection can be sensitive in some corner case. You hit such a case. This extra multiplication caused one optimization to apply more frequently and hit the threshold.